### PR TITLE
Combine clinical effect and biological effect

### PR DESCRIPTION
**Pull request by JiaoJiao123**

### CODE REVIEW

File: core/src/main/java/org/mskcc/cbio/oncokb/util/CacheUtils.java [hunk 1]
```diff
@@ -608,6 +608,22 @@ private static void setAllAlterations() {
...
     }
 
+    public static Set<Evidence> getEvidencesByUUIDs(Set<String> uuids) {
+        synEvidences();
+
+        Set<Evidence> mappedEvis = new HashSet<>();
+        if (uuids != null) {
+            for (Map.Entry<Integer, Set<Evidence>> map : evidences.entrySet()) {
+                for (Evidence evidence : map.getValue()) {
+                    if (uuids.contains(evidence.getUuid())) {
```
Comment (zhx828) [inline core/src/main/java/org/mskcc/cbio/oncokb/util/CacheUtils.java]: Let's also do a null check on evidence as well

File: web/yo/app/scripts/directives/reviewpanel.js [hunk 2]
```diff
@@ -171,15 +179,37 @@ angular.module('oncokbApp')
...
                                 {obj: $scope.mt.description, reviewObj: $scope.mt.description_review, uuid: $scope.mt.description_uuid}];
+
+                            var oncogenicChange = mainUtils.needReview($scope.mt.oncogenic_uuid);
+                            var effectChange = mainUtils.needReview($scope.mt.effect_uuid) || mainUtils.needReview($scope.mt.description_uuid);
+                            if(oncogenicChange && effectChange) {
+                                $scope.bothChanged = true;
+                                DatabaseConnector.getEvidencesByUUIDs([oncogenicUUID, effectUUID], function(result) {
+                                    var resultJSON = JSON.parse(result.status);
+                                    if (_.isArray(resultJSON)) {
+                                        if(resultJSON[0]) {
```
Comment (zhx828) [inline web/yo/app/scripts/directives/reviewpanel.js]: Why not just loop through the resultJSON then?

File: web/yo/app/scripts/directives/reviewpanel.js [hunk 3]
```diff
@@ -171,15 +179,37 @@ angular.module('oncokbApp')
...
+                                            setUpdateTimeEffectSection(resultJSON[0], oncogenicUUID, effectUUID);
+                                        }
+                                        if(resultJSON[1]) {
+                                            setUpdateTimeEffectSection(resultJSON[1], oncogenicUUID, effectUUID);
+                                        }
+                                    }
+                                    specialCases();
```
Comment (zhx828) [inline web/yo/app/scripts/directives/reviewpanel.js]: Why do we go through `specialCases` for ONCOGENIC? I didn't see any switch case inside of that function

Comment (JiaoJiao123) [inline web/yo/app/scripts/directives/reviewpanel.js]: Good catch! we don't need to.

**Pull request merged**